ghga_connector/cli.py

```diff
@@ -99,10 +101,15 @@ def failure(self, message: str):
 
 
 @cli.command()
-def upload(  # noqa C901
+def upload(  # noqa C901, pylint: disable=unused-argument
     api_url: str = typer.Option(..., help="Url to the upload contoller"),
     file_id: str = typer.Option(..., help="The id if the file to upload"),
     file_path: str = typer.Option(..., help="The path to the file to upload"),
+    num_retries: int = typer.Argument(
+        default=MAX_RETRIES,
+        help="Number of times to retry failed part uploads",
+        callback=Retry.set_retries,
```

Comment:
As far as I understand, the callback is rather meant to transform the received argument. Here it is a bit hard to understand what is happening to the num_retries argument and why it is not used.